### PR TITLE
Optimizing CI

### PR DESCRIPTION
**Pull request by Patrick-Pataky**
Currently, the CI takes 15min and 40min, which is huge. The purpose of this PR is to make it really faster !

### CODE REVIEW

**Comment by Patrick-Pataky** [comment]
37 minutes

**Comment by LozanoCyrielle** [comment]
I refactored the tests:
- regrouped all the displayed in one test
- regrouped the basic is working test together

File: app/src/androidTest/java/com/android/partagix/home/HomeTest.kt
Comment (AntoineBut) [inline app/src/androidTest/java/com/android/partagix/home/HomeTest.kt:1]: Good.

File: app/src/androidTest/java/com/android/partagix/home/HomeTest.kt [hunk 1]
```diff
@@ -74,68 +74,25 @@ class HomeTest : TestCase(kaspressoBuilder = Kaspresso.Builder.withComposeSuppor
   }
 
   @Test
-  fun topBarIsDisplayed() = run {
-    ComposeScreen.onComposeScreen<HomeScreen>(composeTestRule) { topBar { assertIsDisplayed() } }
-  }
-
-  @Test
-  fun mainContentIsDisplayed() = run {
+  fun contentIsDisplayed() = run {
```
Comment (AntoineBut) [inline app/src/androidTest/java/com/android/partagix/home/HomeTest.kt:80]: This is a clever addition! Much faster than previous tests.

File: app/src/androidTest/java/com/android/partagix/stamp/StampTest.kt [hunk 2]
```diff
@@ -44,90 +44,40 @@ class StampTest : TestCase(kaspressoBuilder = Kaspresso.Builder.withComposeSuppo
     }
   }
 
-  @Test fun testTest() = run { assert(true) }
```
Comment (AntoineBut) [inline app/src/androidTest/java/com/android/partagix/stamp/StampTest.kt:47]: nice catch

File: gradle.properties [hunk 3]
```diff
@@ -21,4 +21,8 @@ kotlin.code.style=official
 # resources declared in the library itself and none from the library's dependencies,
 # thereby reducing the size of the R class for that library
 android.nonTransitiveRClass=true
-sonar.gradle.skipCompile=true
\ No newline at end of file
+sonar.gradle.skipCompile=true
+
+org.gradle.caching=true
+org.gradle.configuration-cache=true
+org.gradle.parallel=true
```
Comment (AntoineBut) [inline gradle.properties:28]: nice additions

**Review by AntoineBut — APPROVED** [review]
Overall great rework. Now, we need to make sure the entire teams knows how to best combine different tests into a single @Test function, as those have great overhead.

**Pull request merged**